experiment/scripts/tx-parse.py

```diff
@@ -0,0 +1,129 @@
+import struct
+
+class PoppableBytes:
+    def __init__(self, b):
+        self.marker = 0
+        self.excursor = 0
+        self.b = b
+    def pop(self, much):
+        self.marker += much
+        if self.marker > len(self.b):
+            raise
+        return self.b[self.marker-much:self.marker]
+    def peek(self, much):
+        return self.b[self.marker:self.marker+much]
+    def excursion(self):
```

Comment:
Is this standard terminology? I would call it "last". But I dislike that this updates the state of the class. Shouldn't this be done just by pop?

Reply:
Excursion is a term from emacs but not used in the same way here. An excursion here is the concatenation of all pops that have happened since the last excursion was executed. This allows us to do transaction parsing while keeping the parsed parts easily as you can see in the code after it. However this code is not part of the smart contract and is only a helper script I used during development so I'm not too concerned.